### PR TITLE
Refactoring flush and all test cases.

### PR DESCRIPTION
**Pull request by zihejia**

### CODE REVIEW

File: HelloMixpanel/HelloMixpanel/AppDelegate.m [hunk 1]
```diff
@@ -38,17 +38,32 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
...
+    NSString *jsonString = @"[{\"id\": false, \"name\":\"sam\"}]";
+    NSData *data = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+
+    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData: data options:  NSJSONReadingMutableContainers error: &e];
+
+    if (!jsonArray) {
+        NSLog(@"Error parsing JSON: %@", e);
+    } else {
+        for(NSDictionary *item in jsonArray) {
+            NSLog(@"Item: %@", item);
```
Comment (jaredmixpanel) [inline HelloMixpanel/HelloMixpanel/AppDelegate.m:63]: what's the purpose for adding this code?

File: HelloMixpanel/HelloMixpanelTests/MixpanelBaseTests.m [hunk 2]
```diff
@@ -20,59 +20,63 @@ - (void)setUp {
...
-    [self.mixpanel reset];
-    [self waitForMixpanelQueues];
-    [self deleteOptOutSettingsWithMixpanelInstance:self.mixpanel];
+- (NSArray *)peopleQueue:(NSString *)token {
+    return [[[MixpanelPersistence alloc] initWithToken:token] loadEntitiesInBatch:PersistenceTypePeople];
+}
 
-    self.mixpanel = nil;
+- (NSArray *)unIdentifiedPeopleQueue:(NSString *)token {
+    return [[[MixpanelPersistence alloc] initWithToken:token] loadEntitiesInBatch:PersistenceTypePeople flag:YES];
```
Comment (jaredmixpanel) [inline HelloMixpanel/HelloMixpanelTests/MixpanelBaseTests.m:55]: `flag: UnIdentifiedFlag`

File: HelloMixpanel/HelloMixpanel_tvOS/ViewController.m [hunk 3]
```diff
@@ -22,7 +22,7 @@ - (IBAction)setPeopleProperties:(id)sender {
     
     // Mixpanel People requires that you explicitly set a distinct ID for the current user. In this case,
     // we're using the automatically generated distinct ID from event tracking.
-    [mixpanel identify:mixpanel.distinctId];
+   // [mixpanel identify:mixpanel.distinctId];
```
Comment (jaredmixpanel) [inline HelloMixpanel/HelloMixpanel_tvOS/ViewController.m:25]: why'd we comment this out?

File: Sources/MixpanelIdentity.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ 
+ 
+ NSString *prefix = [NSString stringWithFormat:@"%@-%@", kDefaultKeyPrefix, apiToken];
+ NSString *distinctId = [defaults stringForKey:[NSString stringWithFormat:@"%@%@", prefix, kDefaultKeyDistinctId]];
+ NSString *peopleDistinctId = [defaults stringForKey:[NSString stringWithFormat:@"%@%@", prefix, kDefaultKeyPeopleDistinctId]];
+ NSString *anonymousId = [defaults stringForKey:[NSString stringWithFormat:@"%@%@", prefix, kDefaultKeyAnonymousId]];
+ NSString *userId = [defaults stringForKey:[NSString stringWithFormat:@"%@%@", prefix, kDefaultKeyUserId]];
+ NSString *alias = [defaults stringForKey:[NSString stringWithFormat:@"%@%@", prefix, kDefaultKeyAlias]];
+ NSString *hadPersistedDistinctId = [defaults stringForKey:[NSString stringWithFormat:@"%@%@", prefix, kDefaultKeyHadPersistedDistinctId]];
+
```
Comment (jaredmixpanel) [inline Sources/MixpanelIdentity.h:21]: we can delete this, right?

File: Sources/MixpanelPersistence.m [hunk 5]
```diff
@@ -35,48 +33,63 @@ @implementation MixpanelPersistence : NSObject
...
     return self;
 }
 
-- (void)saveEntity:(NSDictionary *)entity type:(NSString *)type flag:(bool)flag {
-    NSError *error;
-    NSData *data = [NSJSONSerialization dataWithJSONObject:entity options:kNilOptions error:&error];
-    if (!error) {
+
+- (void)saveEntity:(NSDictionary *)entity type:(NSString *)type {
+    [self saveEntity:entity type:type flag:NO];
```
Comment (jaredmixpanel) [inline Sources/MixpanelPersistence.m:47]: `flag: IdentifiedFlag`

File: Sources/MixpanelPersistence.m [hunk 6]
```diff
@@ -35,48 +33,63 @@ @implementation MixpanelPersistence : NSObject
...
+- (void)saveEntities:(NSArray *)entities type:(NSString *)type flag:(BOOL)flag {
     for (NSDictionary *entity in entities) {
         [self saveEntity:entity type:type flag:flag];
     }
 }
 
-- (NSArray *)loadEntitiesInBatch:(NSString *)type batchSize:(int)batchSize flag:(bool)flag {
-    return [self.mpdb readRows:type numRows:batchSize flag:flag];
+- (NSArray *)loadEntitiesInBatch:(NSString *)type {
+    return [self loadEntitiesInBatch:type flag:NO];
```
Comment (jaredmixpanel) [inline Sources/MixpanelPersistence.m:64]: `flag:IdentifiedFlag`

File: Sources/MixpanelPersistence.m [hunk 7]
```diff
@@ -35,48 +33,63 @@ @implementation MixpanelPersistence : NSObject
...
+    return entities;
 }
 
 - (void)removeEntitiesInBatch:(NSString *)type ids:(NSArray *)ids {
     [self.mpdb deleteRows:type ids:ids];
 }
 
 - (void)identifyPeople {
-    [self.mpdb updateRowsFlag:@"people" newFlag:false];
+    [self.mpdb updateRowsFlag:@"people" newFlag:NO];
```
Comment (jaredmixpanel) [inline Sources/MixpanelPersistence.m:83]: `flag:IdentifiedFlag`

File: Sources/MixpanelIdentity.h (hunk 4 above)
Comment (jaredmixpanel) [reply Sources/MixpanelIdentity.h:21]: Ah, nvm, I see this is a peek at `loadIdentity:`

**Review by jaredmixpanel — APPROVED** [review]
a few notes, but LGTM!

File: HelloMixpanel/HelloMixpanel/AppDelegate.m (hunk 1 above)
Comment (zihejia) [reply HelloMixpanel/HelloMixpanel/AppDelegate.m:63]: whoops, will delete.

File: HelloMixpanel/HelloMixpanel_tvOS/ViewController.m (hunk 3 above)
Comment (zihejia) [reply HelloMixpanel/HelloMixpanel_tvOS/ViewController.m:25]: that was for a temp test, will revert it back.

**Pull request merged**